using-timescaledb/continuous-aggregates.md

```diff
@@ -248,6 +248,124 @@ hypertable with a continuous aggregate.
+the hypertable using
+[`set_integer_now_func`][api-set-integer-now-func]. The function can
+be defined as a normal Postgres function, but need to be
+[`STABLE`][pg-func-stable], take no arguments, and should return an
+integer of the same type as the time column in the table. In our case,
+this should suffice:
+
+```sql
+CREATE OR REPLACE FUNCTION current_microfortnight() RETURNS BIGINT
+LANGUAGE SQL STABLE AS $$
+	SELECT CAST(1209600 * EXTRACT(EPOCH FROM CURRENT_TIMESTAMP) / 1000000 AS BIGINT)
+$$;
+
+SELECT set_integer_now_func('devices', 'current_microfortnight');
+```
+
+Once the replacement for current timestamp has been set up, you can
+define the continuous aggregate for the `devices` table.
+
+```sql
+CREATE VIEW devices_summary
+WITH (timescaledb.continuous,
+      timescaledb.refresh_interval = '1 minute',
+      timescaledb.refresh_lag = '-500') AS
+SELECT time_bucket('500', time) AS bucket,
+       avg(cpu_usage) AS avg_cpu,
+       avg(disk_usage) AS avg_disk
+     FROM devices
+    GROUP BY bucket;
+```
+
+You can now insert some rows to check if the aggregation works as
+expected. In this case, we are using the `tablefunc` extension to
+generate a Gaussian distribution of CPU usage and a cumulative growth
+of disk usage.
+
+```sql
+CREATE EXTENSION tablefunc;
+
+INSERT INTO devices(time, cpu_usage, disk_usage)
+SELECT time,
+       normal_rand(1,70,10) AS cpu_usage,
+	   normal_rand(1,2,1) * (row_number() over()) AS disk_usage
+  FROM generate_series(1,10000) AS time;
+```
+
+You can now check that the view contain the correct data.
+
+```sql
+postgres=# SELECT * FROM devices_summary ORDER BY bucket;
+ bucket |       avg_cpu       |       avg_disk
+--------+---------------------+----------------------
+      0 | 63.0000000000000000 |   6.0000000000000000
+      5 | 69.8000000000000000 |   9.6000000000000000
+     10 | 70.8000000000000000 |  24.0000000000000000
+     15 | 75.8000000000000000 |  37.6000000000000000
+     20 | 71.6000000000000000 |  26.8000000000000000
+     25 | 67.6000000000000000 |  56.0000000000000000
+     30 | 68.8000000000000000 |  90.2000000000000000
+     35 | 71.6000000000000000 |  88.8000000000000000
+     40 | 66.4000000000000000 |  81.2000000000000000
+     45 | 68.2000000000000000 | 106.0000000000000000
+     50 | 72.2000000000000000 | 105.0000000000000000
+     55 | 66.2000000000000000 | 138.8000000000000000
+     60 | 61.6000000000000000 | 131.4000000000000000
+     65 | 64.2000000000000000 | 136.4000000000000000
+     70 | 72.0000000000000000 | 114.8000000000000000
+     75 | 64.4000000000000000 | 155.2000000000000000
+     80 | 65.8000000000000000 | 175.2000000000000000
+     85 | 68.0000000000000000 | 156.8000000000000000
+     90 | 71.6000000000000000 | 151.4000000000000000
+     95 | 68.6000000000000000 | 136.2000000000000000
+(20 rows)
+```
+
+If you want to refresh the view immediately, you can do it in the
+normal manner:
+
+```sql
+REFRESH MATERIALIZED VIEW devices_summary;
```

Comment:
Note that this might be confusing because apparently this is designed to not be called directly. Also, it won't always refresh the view as one would expect (depends on your view settings). I guess it works in this case because you configured a `refresh_lag` of `-bucket`, which is not the recommended (or default) setting since it makes the agg inefficient.

I know this is not intuitive, but that's also why I am wondering if we shouldn't mention manual refresh since it currently doesn't do what you expect in all cases.

Reply:
It is already in the manual as a tip above, so we should not shy from mentioning it. If it does not behave as expected in all cases, we should document this, or fix the behavior, but I do not think that omitting mentioning it is a helpful for the users.

Also, I do not think that refresh lag affects performance, see comment above, but I might have misunderstood how it works.

Reply:
The docs says that an aggressive refresh_lag affects insert performance and increases write amplification. (See above).

If refresh_lag would not affect performance, then there is no reason for that parameter to exist at all since one would always prefer no lag.

Reply:
AFAICT, write amplification does not automatically lead to lower performance, but it does increase utilization. However, I still think that we should keep the `REFRESH MATERIALIZED VIEW` given that it is already discussed elsewhere in the documentation.

Reply:
Ok, let's keep `REFRESH MATERIALIZED VIEW`. Fwiw, I think this command _should_ be a first-class command for users since it is for regular views. It's just that I've heard some people say it is not supposed to be used (and it  prints a lot of cryptic info) and semantics of it is currently confusing since it doesn't completely refresh the view like you would expect.

Reply:
@mkindahl I think this is the REFRESH ... that you referring to:
_:TIP: Most times the continuous aggregate view will be updated by the background job; however, if you would like to run it yourself, you may use the REFRESH MATERIALIZED VIEW command._

should be rephrased as : 
:TIP: The continuous aggregate view will be updated by the background job; however, if you would like to verify if the view is working correctly, you may use the REFRESH MATERIALIZED VIEW command.

Reply:
Fixed the tip, but if `alter_job_schedule` is going to be used, you need to dig out the `job_id`, which is not obvious how to do, so I would say that we either keep the `REFRESH` even though it is not perfect, or remove it entirely.

I think it is better to keep it since it does ensure that the view shows *something*, even it if is not perfect. The usage in the tutorial is more reminiscent of what you would do when debugging (just see that it works, don't really care about the exact figures).

Reply:
If you are keeping REFRESH, please add a note that this is generally how we manually check if the view is working and not running into any errors.The REFRESH MATERIALIZED VIEW will behave like  a _single run_ of the (automatically run) background worker job. (I don't think we actually say this anywhere else).
Please modify the verbiage around this paragraph because it could be interpreted as the entire view is getting updated

_If you want to refresh the
continuous aggregate view immediately, you can use [`REFRESH
MATERIALIZED VIEW`][api-refresh-continuous-aggs]:_

Reply:
As per my other comment, I think saying that you can use a manual refresh as a method of checking if things are working is highly misleading. A background job could be failing or missing even if a manual refresh works.

Reply:
@erminator If the manual refresh works, the background job that does the automatic refresh should work as well.i.e to say if manual refresh does not work => bgw job will fail. (Yes, there are other cases where bgw jobs  could fail but they are related to the job scheduling and not to the materialization itself). We have a materialization bug if the bgw job materialization fails but the refresh succeeds

Reply:
Removed all text mentioning the `REFRESH MATERIALIZED VIEW`.